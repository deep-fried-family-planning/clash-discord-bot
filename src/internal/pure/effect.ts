import {Order} from 'effect';


export {
    Effect as E,
    Console as CSL,
    Data as D,
    DateTime as DT,
    Redacted as RDT,
    Config as CFG,
    Context as CTX,
    Layer as L,
    Cache as C,
    Logger as Logger,
    LogLevel,
    LogSpan,
    Schema as S,
    Brand as B,
    RateLimiter as RL,
    Metric as Metric,
    Order as ORD,
    Option as O,
    Equal as EQ,

    Cron,

    pipe,
    pipe as p,
    flow,
    flow as f,

} from 'effect';


export {
    gen as g,

} from 'effect/Effect';


export const ORDS = Order.string;
export const ORDSR = Order.reverse(Order.string);
export const ORDN = Order.number;
export const ORDNR = Order.reverse(Order.number);


export * as Ar from 'effect/Array'; // collection array
export * as Kv from 'effect/Record'; // collection key-value
