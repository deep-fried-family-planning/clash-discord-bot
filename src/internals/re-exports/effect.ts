export {
    Effect as E,

    Config as Cfg,
    Redacted as Rdt,

    Console as CSL,

    Data as D,
    DateTime as DT,

    Redacted as RDT,
    Config as CFG,
    Context as CTX,
    Layer as L,

    Cache as C,

    Logger as Logger,
    LogLevel,
    LogSpan,

    Schema as S,

    Brand as B,

    RateLimiter as RL,

    Metric as Metric,

    flow,
    pipe,

} from 'effect';
