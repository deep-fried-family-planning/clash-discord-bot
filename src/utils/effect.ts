export {
    Effect as E,

    Config as Cfg,
    Redacted as Rdt,

    Layer as L,

    Logger as Logger,
    LogLevel,
    LogSpan,

    RateLimiter as RL,

    Metric as Metric,

    flow,
    pipe,

} from 'effect';
