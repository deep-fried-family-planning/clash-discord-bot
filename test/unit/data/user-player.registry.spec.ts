import {DataTag} from '#src/data/constants/index.ts';
import {UserPlayerRegistry, UserRegistry} from '#src/data/index.ts';
import {mockCoc, mockCocLayer} from '#unit/data/mock-coc.ts';
import {mockDb, mockDbLayer} from '#unit/data/mock-db.ts';
import {it, describe} from '@effect/vitest';
import * as DateTime from 'effect/DateTime';
import * as E from 'effect/Effect';
import {pipe} from 'effect/Function';

describe('given caller user is not registered', () => {
  it.effect('when registering user player', E.fn(function* () {
    mockDb.get
      .mockReturnValueOnce(E.succeed({Item: undefined}));

    const actual = yield* pipe(
      UserPlayerRegistry.register({
        caller_id : 'user',
        player_tag: 'player',
        api_token : 'api_token',
        payload   : {
          account_type: 'main',
        },
      }),
      E.provide(mockDbLayer),
      E.provide(mockCocLayer),
      E.catchAll((cause) => E.succeed(cause)),
    );

    expect(actual).toMatchInlineSnapshot(`[NoSuchElementException]`);
    expect(mockCoc.verifyPlayerToken).not.toBeCalled();
    expect(mockCoc.getPlayer).not.toBeCalled();
    expect(mockDb.get.mock.calls[0][0].Key).toMatchSnapshot();
    expect(mockDb.query).not.toBeCalled();
    expect(mockDb.put).not.toBeCalled();
    expect(mockDb.delete).not.toBeCalled();
  }));
});

describe('given new user player registration', () => {
  it.effect('when registering user player', E.fn(function* () {
    mockDb.get
      .mockReturnValueOnce(E.succeed({
        Item: {
          _tag           : DataTag.USER,
          pk             : 'u-user',
          sk             : 'now',
          version        : 0,
          created        : DateTime.unsafeMake(0).pipe(DateTime.format()),
          updated        : DateTime.unsafeMake(0).pipe(DateTime.format()),
          gsi_all_user_id: 'u-user',
          timezone       : 'America/New_York',
        },
      }));

    mockDb.query
      .mockReturnValueOnce(E.succeed({Items: []}));

    mockCoc.verifyPlayerToken
      .mockReturnValueOnce(E.succeed(true));

    mockCoc.getPlayer
      .mockReturnValueOnce(E.succeed({
        name: 'PlayerName',
      }));

    const actual = yield* pipe(
      UserPlayerRegistry.register({
        caller_id : 'user',
        player_tag: 'player',
        api_token : 'api_token',
        payload   : {
          account_type: 'main',
        },
      }),
      E.provide(mockDbLayer),
      E.provide(mockCocLayer),
    );

    expect(actual).toMatchInlineSnapshot(`
      {
        "description": "Success",
      }
    `);
    expect(mockCoc.verifyPlayerToken).toBeCalled();
    expect(mockCoc.verifyPlayerToken.mock.calls[0]).toMatchSnapshot();
    expect(mockCoc.getPlayer).toBeCalled();
    expect(mockCoc.getPlayer.mock.calls[0]).toMatchSnapshot();
    expect(mockDb.get).toBeCalled();
    expect(mockDb.get.mock.calls[0][0].Key).toMatchSnapshot();
    expect(mockDb.query).toBeCalled();
    expect(mockDb.query.mock.calls[0][0].KeyConditionExpression).toMatchSnapshot();
    expect(mockDb.put).toBeCalled();
    expect(mockDb.put.mock.calls[0][0].Item).toMatchSnapshot();
    expect(mockDb.delete).not.toBeCalled();
  }));

  describe('given invalid api_token', () => {
    it.effect('when caller user is not registered', E.fn(function* () {
      mockDb.get
        .mockReturnValueOnce(E.succeed({
          Item: {
            _tag           : DataTag.USER,
            pk             : 'u-user',
            sk             : 'now',
            version        : 0,
            created        : DateTime.unsafeMake(0).pipe(DateTime.format()),
            updated        : DateTime.unsafeMake(0).pipe(DateTime.format()),
            gsi_all_user_id: 'u-user',
            timezone       : 'America/New_York',
          },
        }));

      mockDb.query
        .mockReturnValueOnce(E.succeed({Items: []}));

      mockCoc.verifyPlayerToken.mockReturnValueOnce(E.succeed(false));

      const actual = yield* pipe(
        UserPlayerRegistry.register({
          caller_id : 'user',
          player_tag: 'player',
          api_token : 'api_token',
          payload   : {
            account_type: 'main',
          },
        }),
        E.catchAll((cause) => E.succeed(cause)),
        E.provide(mockDbLayer),
        E.provide(mockCocLayer),
      );

      expect(actual).toMatchInlineSnapshot(`[RegistryUserError: Invalid player token.]`);
      expect(mockCoc.verifyPlayerToken).toBeCalled();
      expect(mockCoc.verifyPlayerToken.mock.calls[0]).toMatchSnapshot();
      expect(mockCoc.getPlayer).not.toBeCalled();
      expect(mockDb.get).toBeCalled();
      expect(mockDb.get.mock.calls[0][0].Key).toMatchSnapshot();
      expect(mockDb.query).not.toBeCalled();
      expect(mockDb.put).not.toBeCalled();
      expect(mockDb.delete).not.toBeCalled();
    }));
  });
});

describe('given user player is already registered', () => {
  it.effect('when registering user player', E.fn(function* () {
    mockDb.get
      .mockReturnValueOnce(E.succeed({
        Item: {
          _tag           : DataTag.USER,
          pk             : 'u-user',
          sk             : 'now',
          version        : 0,
          created        : DateTime.unsafeMake(0).pipe(DateTime.format()),
          updated        : DateTime.unsafeMake(0).pipe(DateTime.format()),
          gsi_all_user_id: 'u-user',
          timezone       : 'America/New_York',
        },
      }));

    mockDb.query
      .mockReturnValueOnce(E.succeed({Items: [{
          _tag          : 'UserPlayer',
          account_type  : 'main',
          created       : '1970-01-01T00:00:00.000Z',
          gsi_player_tag: 'p-player',
          gsi_user_id   : 'u-user2',
          name          : 'PlayerName',
          pk            : 'u-user2',
          sk            : 'p-player',
          updated       : '1970-01-01T00:00:00.000Z',
          verification  : 2,
          version       : 0,
        }]}));

    mockCoc.verifyPlayerToken
      .mockReturnValueOnce(E.succeed(true));

    mockCoc.getPlayer
      .mockReturnValueOnce(E.succeed({
        name: 'PlayerName',
      }));

    const actual = yield* pipe(
      UserPlayerRegistry.register({
        caller_id : 'user',
        player_tag: 'player',
        api_token : 'api_token',
        payload   : {
          account_type: 'main',
        },
      }),
      E.provide(mockDbLayer),
      E.provide(mockCocLayer),
    );

    expect(actual).toMatchInlineSnapshot(`
      {
        "description": "Success",
      }
    `);
    expect(mockCoc.verifyPlayerToken).toBeCalled();
    expect(mockCoc.verifyPlayerToken.mock.calls[0]).toMatchSnapshot();
    expect(mockCoc.getPlayer).toBeCalled();
    expect(mockCoc.getPlayer.mock.calls[0]).toMatchSnapshot();
    expect(mockDb.get).toBeCalled();
    expect(mockDb.get.mock.calls[0][0].Key).toMatchSnapshot();
    expect(mockDb.query).toBeCalled();
    expect(mockDb.query.mock.calls[0][0].KeyConditionExpression).toMatchSnapshot();
    expect(mockDb.put).toBeCalled();
    expect(mockDb.put.mock.calls[0][0].Item).toMatchSnapshot();
    expect(mockDb.delete).toBeCalled();
    expect(mockDb.delete.mock.calls[0][0].Key).toMatchSnapshot();
  }));
});

describe('given caller is attempting admin registration', () => {
  it.effect('when admin registering user player', E.fn(function* () {
    mockDb.get
      .mockReturnValueOnce(E.succeed({
        Item: {
          _tag           : DataTag.USER,
          pk             : 'u-user',
          sk             : 'now',
          version        : 0,
          created        : DateTime.unsafeMake(0).pipe(DateTime.format()),
          updated        : DateTime.unsafeMake(0).pipe(DateTime.format()),
          gsi_all_user_id: 'u-user',
          timezone       : 'America/New_York',
        },
      }))
      .mockReturnValueOnce(E.succeed({
        Item: {
          _tag             : 'Server',
          admin            : 'admin',
          created          : '1970-01-01T00:00:00.000Z',
          gsi_all_server_id: 's-guild',
          pk               : 's-guild',
          sk               : 'now',
          updated          : '1970-01-01T00:00:00.000Z',
          version          : 0,
        },
      }));

    mockDb.query
      .mockReturnValueOnce(E.succeed({Items: [{
          _tag          : 'UserPlayer',
          account_type  : 'main',
          created       : '1970-01-01T00:00:00.000Z',
          gsi_player_tag: 'p-player',
          gsi_user_id   : 'u-user2',
          name          : 'PlayerName',
          pk            : 'u-user2',
          sk            : 'p-player',
          updated       : '1970-01-01T00:00:00.000Z',
          verification  : 1,
          version       : 0,
        }]}));

    mockCoc.verifyPlayerToken
      .mockReturnValueOnce(E.succeed(true));

    mockCoc.getPlayer
      .mockReturnValueOnce(E.succeed({
        name: 'PlayerName',
      }));

    const actual = yield* pipe(
      UserPlayerRegistry.register({
        guild_id    : 'guild',
        caller_id   : 'user',
        player_tag  : 'player',
        caller_roles: ['admin'],
        target_id   : 'user2',
        payload     : {
          account_type: 'main',
        },
      }),
      E.provide(mockDbLayer),
      E.provide(mockCocLayer),
    );

    expect(actual).toMatchInlineSnapshot(`
      {
        "description": "Success",
      }
    `);
    expect(mockCoc.verifyPlayerToken).not.toBeCalled();
    expect(mockCoc.getPlayer).toBeCalled();
    expect(mockCoc.getPlayer.mock.calls[0]).toMatchSnapshot();
    expect(mockDb.get).toBeCalled();
    expect(mockDb.get.mock.calls[0][0].Key).toMatchSnapshot();
    expect(mockDb.get.mock.calls[1][0].Key).toMatchSnapshot();
    expect(mockDb.query).toBeCalled();
    expect(mockDb.query.mock.calls[0][0].KeyConditionExpression).toMatchSnapshot();
    expect(mockDb.put).toBeCalled();
    expect(mockDb.put.mock.calls[0][0].Item).toMatchSnapshot();
    expect(mockDb.delete).toBeCalled();
    expect(mockDb.delete.mock.calls[0][0].Key).toMatchSnapshot();
  }));

  describe('given player is already registered at higher verification level', () => {
    it.effect('when admin registering user player', E.fn(function* () {
      mockDb.get
        .mockReturnValueOnce(E.succeed({
          Item: {
            _tag           : DataTag.USER,
            pk             : 'u-user',
            sk             : 'now',
            version        : 0,
            created        : DateTime.unsafeMake(0).pipe(DateTime.format()),
            updated        : DateTime.unsafeMake(0).pipe(DateTime.format()),
            gsi_all_user_id: 'u-user',
            timezone       : 'America/New_York',
          },
        }))
        .mockReturnValueOnce(E.succeed({
          Item: {
            _tag             : 'Server',
            admin            : 'admin',
            created          : '1970-01-01T00:00:00.000Z',
            gsi_all_server_id: 's-guild',
            pk               : 's-guild',
            sk               : 'now',
            updated          : '1970-01-01T00:00:00.000Z',
            version          : 0,
          },
        }));

      mockDb.query
        .mockReturnValueOnce(E.succeed({Items: [{
            _tag          : 'UserPlayer',
            account_type  : 'main',
            created       : '1970-01-01T00:00:00.000Z',
            gsi_player_tag: 'p-player',
            gsi_user_id   : 'u-user2',
            name          : 'PlayerName',
            pk            : 'u-user2',
            sk            : 'p-player',
            updated       : '1970-01-01T00:00:00.000Z',
            verification  : 2,
            version       : 0,
          }]}));

      mockCoc.verifyPlayerToken
        .mockReturnValueOnce(E.succeed(true));

      mockCoc.getPlayer
        .mockReturnValueOnce(E.succeed({
          name: 'PlayerName',
        }));

      const actual = yield* pipe(
        UserPlayerRegistry.register({
          guild_id    : 'guild',
          caller_id   : 'user',
          player_tag  : 'player',
          caller_roles: ['admin'],
          target_id   : 'user2',
          payload     : {
            account_type: 'main',
          },
        }),
        E.provide(mockDbLayer),
        E.provide(mockCocLayer),
        E.catchAll((cause) => E.succeed(cause)),
      );

      expect(actual).toMatchInlineSnapshot(`[RegistryAdminError: You are not authorized to update the registration of this player account.]`);
      expect(mockCoc.verifyPlayerToken).not.toBeCalled();
      expect(mockCoc.getPlayer).toBeCalled();
      expect(mockCoc.getPlayer.mock.calls[0]).toMatchSnapshot();
      expect(mockDb.get).toBeCalled();
      expect(mockDb.get.mock.calls[0][0].Key).toMatchSnapshot();
      expect(mockDb.get.mock.calls[1][0].Key).toMatchSnapshot();
      expect(mockDb.query).toBeCalled();
      expect(mockDb.query.mock.calls[0][0].KeyConditionExpression).toMatchSnapshot();
      expect(mockDb.put).not.toBeCalled();
      expect(mockDb.delete).not.toBeCalled();
    }));
  });

  describe('given caller server is not registered', () => {
    it.effect('when admin registering user player', E.fn(function* () {
      mockDb.get
        .mockReturnValueOnce(E.succeed({
          Item: {
            _tag           : DataTag.USER,
            pk             : 'u-user',
            sk             : 'now',
            version        : 0,
            created        : DateTime.unsafeMake(0).pipe(DateTime.format()),
            updated        : DateTime.unsafeMake(0).pipe(DateTime.format()),
            gsi_all_user_id: 'u-user',
            timezone       : 'America/New_York',
          },
        }))
        .mockReturnValueOnce(E.succeed({
          Item: undefined,
        }));

      const actual = yield* pipe(
        UserPlayerRegistry.register({
          guild_id    : 'guild',
          caller_id   : 'user',
          player_tag  : 'player',
          caller_roles: ['admin'],
          target_id   : 'user2',
          payload     : {
            account_type: 'main',
          },
        }),
        E.provide(mockDbLayer),
        E.provide(mockCocLayer),
        E.catchAll((cause) => E.succeed(cause)),
      );

      expect(actual).toMatchInlineSnapshot(`[NoSuchElementException]`);
      expect(mockCoc.verifyPlayerToken).not.toBeCalled();
      expect(mockCoc.getPlayer).not.toBeCalled();
      expect(mockDb.get).toBeCalled();
      expect(mockDb.query).not.toBeCalled();
      expect(mockDb.put).not.toBeCalled();
      expect(mockDb.delete).not.toBeCalled();
    }));
  });

  describe('given caller is not a server admin', () => {
    it.effect('when admin registering user player', E.fn(function* () {
      mockDb.get
        .mockReturnValueOnce(E.succeed({
          Item: {
            _tag           : DataTag.USER,
            pk             : 'u-user',
            sk             : 'now',
            version        : 0,
            created        : DateTime.unsafeMake(0).pipe(DateTime.format()),
            updated        : DateTime.unsafeMake(0).pipe(DateTime.format()),
            gsi_all_user_id: 'u-user',
            timezone       : 'America/New_York',
          },
        }))
        .mockReturnValueOnce(E.succeed({
          Item: {
            _tag             : 'Server',
            admin            : 'admin',
            created          : '1970-01-01T00:00:00.000Z',
            gsi_all_server_id: 's-guild',
            pk               : 's-guild',
            sk               : 'now',
            updated          : '1970-01-01T00:00:00.000Z',
            version          : 0,
          },
        }));

      const actual = yield* pipe(
        UserPlayerRegistry.register({
          guild_id    : 'guild',
          caller_id   : 'user',
          player_tag  : 'player',
          caller_roles: [],
          target_id   : 'user2',
          payload     : {
            account_type: 'main',
          },
        }),
        E.provide(mockDbLayer),
        E.provide(mockCocLayer),
        E.catchAll((cause) => E.succeed(cause)),
      );

      expect(actual).toMatchInlineSnapshot(`[RegistryAdminError: You are not authorized to register a player account for this server.]`);
      expect(mockCoc.verifyPlayerToken).not.toBeCalled();
      expect(mockCoc.getPlayer).not.toBeCalled();
      expect(mockDb.get).toBeCalled();
      expect(mockDb.query).not.toBeCalled();
      expect(mockDb.put).not.toBeCalled();
      expect(mockDb.delete).not.toBeCalled();
    }));
  });
});
