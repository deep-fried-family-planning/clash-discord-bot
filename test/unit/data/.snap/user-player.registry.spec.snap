// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`given caller is attempting admin registration > given caller is not a server admin > when admin registering user player 1`] = `[RegistryAdminError: You are not authorized to update the registration of this player account.]`;

exports[`given caller is attempting admin registration > given caller is not a server admin > when admin registering user player 2`] = `
[
  "player",
]
`;

exports[`given caller is attempting admin registration > given caller is not a server admin > when admin registering user player 3`] = `
{
  "pk": "u-user",
  "sk": "now",
}
`;

exports[`given caller is attempting admin registration > given caller is not a server admin > when admin registering user player 4`] = `
{
  "pk": "s-guild",
  "sk": "now",
}
`;

exports[`given caller is attempting admin registration > given caller is not a server admin > when admin registering user player 5`] = `"gsi_player_tag = :gsi_player_tag"`;

exports[`given caller is attempting admin registration > given caller server is not registered > when admin registering user player 1`] = `[RegistryAdminError: You are not authorized to update the registration of this player account.]`;

exports[`given caller is attempting admin registration > given caller server is not registered > when admin registering user player 2`] = `
[
  "player",
]
`;

exports[`given caller is attempting admin registration > given caller server is not registered > when admin registering user player 3`] = `
{
  "pk": "u-user",
  "sk": "now",
}
`;

exports[`given caller is attempting admin registration > given caller server is not registered > when admin registering user player 4`] = `
{
  "pk": "s-guild",
  "sk": "now",
}
`;

exports[`given caller is attempting admin registration > given caller server is not registered > when admin registering user player 5`] = `"gsi_player_tag = :gsi_player_tag"`;

exports[`given caller is attempting admin registration > given player is already registered at higher verification level > when admin registering user player 1`] = `[RegistryAdminError: You are not authorized to update the registration of this player account.]`;

exports[`given caller is attempting admin registration > given player is already registered at higher verification level > when admin registering user player 2`] = `
[
  "player",
]
`;

exports[`given caller is attempting admin registration > given player is already registered at higher verification level > when admin registering user player 3`] = `
{
  "pk": "u-user",
  "sk": "now",
}
`;

exports[`given caller is attempting admin registration > given player is already registered at higher verification level > when admin registering user player 4`] = `
{
  "pk": "s-guild",
  "sk": "now",
}
`;

exports[`given caller is attempting admin registration > given player is already registered at higher verification level > when admin registering user player 5`] = `"gsi_player_tag = :gsi_player_tag"`;

exports[`given caller is attempting admin registration > when admin registering user player 1`] = `
{
  "description": "Success",
}
`;

exports[`given caller is attempting admin registration > when admin registering user player 2`] = `
[
  "player",
]
`;

exports[`given caller is attempting admin registration > when admin registering user player 3`] = `
{
  "pk": "u-user",
  "sk": "now",
}
`;

exports[`given caller is attempting admin registration > when admin registering user player 4`] = `
{
  "pk": "s-guild",
  "sk": "now",
}
`;

exports[`given caller is attempting admin registration > when admin registering user player 5`] = `"gsi_player_tag = :gsi_player_tag"`;

exports[`given caller is attempting admin registration > when admin registering user player 6`] = `
{
  "_tag": "UserPlayer",
  "account_type": "main",
  "created": "1970-01-01T00:00:00.000Z",
  "gsi_player_tag": "p-player",
  "gsi_user_id": "u-user2",
  "name": "PlayerName",
  "pk": "u-user2",
  "sk": "p-player",
  "updated": "1970-01-01T00:00:00.000Z",
  "verification": 1,
  "version": 0,
}
`;

exports[`given caller is attempting admin registration > when admin registering user player 7`] = `
{
  "pk": "u-user2",
  "sk": "p-player",
}
`;

exports[`given caller user is not registered > when registering user player 1`] = `[NoSuchElementException]`;

exports[`given caller user is not registered > when registering user player 2`] = `
{
  "pk": "u-user",
  "sk": "now",
}
`;

exports[`given new user player registration > given invalid api_token > when caller user is not registered 1`] = `[RegistryUserError: Invalid player token.]`;

exports[`given new user player registration > given invalid api_token > when caller user is not registered 2`] = `
[
  "player",
  "api_token",
]
`;

exports[`given new user player registration > given invalid api_token > when caller user is not registered 3`] = `
{
  "pk": "u-user",
  "sk": "now",
}
`;

exports[`given new user player registration > when registering user player 1`] = `
{
  "description": "Success",
}
`;

exports[`given new user player registration > when registering user player 2`] = `
[
  "player",
  "api_token",
]
`;

exports[`given new user player registration > when registering user player 3`] = `
[
  "player",
]
`;

exports[`given new user player registration > when registering user player 4`] = `
{
  "pk": "u-user",
  "sk": "now",
}
`;

exports[`given new user player registration > when registering user player 5`] = `"gsi_player_tag = :gsi_player_tag"`;

exports[`given new user player registration > when registering user player 6`] = `
{
  "_tag": "UserPlayer",
  "account_type": "main",
  "created": "1970-01-01T00:00:00.000Z",
  "gsi_player_tag": "p-player",
  "gsi_user_id": "u-user",
  "name": "PlayerName",
  "pk": "u-user",
  "sk": "p-player",
  "updated": "1970-01-01T00:00:00.000Z",
  "verification": 2,
  "version": 0,
}
`;

exports[`given user player is already registered > when registering user player 1`] = `
{
  "description": "Success",
}
`;

exports[`given user player is already registered > when registering user player 2`] = `
[
  "player",
  "api_token",
]
`;

exports[`given user player is already registered > when registering user player 3`] = `
[
  "player",
]
`;

exports[`given user player is already registered > when registering user player 4`] = `
{
  "pk": "u-user",
  "sk": "now",
}
`;

exports[`given user player is already registered > when registering user player 5`] = `"gsi_player_tag = :gsi_player_tag"`;

exports[`given user player is already registered > when registering user player 6`] = `
{
  "_tag": "UserPlayer",
  "account_type": "main",
  "created": "1970-01-01T00:00:00.000Z",
  "gsi_player_tag": "p-player",
  "gsi_user_id": "u-user",
  "name": "PlayerName",
  "pk": "u-user",
  "sk": "p-player",
  "updated": "1970-01-01T00:00:00.000Z",
  "verification": 2,
  "version": 0,
}
`;

exports[`given user player is already registered > when registering user player 7`] = `
{
  "pk": "u-user2",
  "sk": "p-player",
}
`;
